src/classes/Utility.py

```diff
@@ -36,100 +36,127 @@
         input('')
 
     # Save last screened result to pickle file
-    def setLastScreenedResults(df):
+    def setLastScreenedResults(self, df):
         try:
             df.sort_values(by=['Stock'], ascending=True, inplace=True)
             df.to_pickle(lastScreened)
         except IOError:
-            input(colorText.BOLD + colorText.FAIL + '[+] Failed to save recently screened result table on disk! Skipping..' + colorText.END)
+            input(colorText.BOLD + colorText.FAIL +
+                  '[+] Failed to save recently screened result table on disk! Skipping..' + colorText.END)
 
     # Load last screened result to pickle file
-    def getLastScreenedResults():
+    def getLastScreenedResults(self):
         try:
             df = pd.read_pickle(lastScreened)
-            print(colorText.BOLD + colorText.GREEN + '\n[+] Showing recently screened results..\n' + colorText.END)
+            print(colorText.BOLD + colorText.GREEN +
+                  '\n[+] Showing recently screened results..\n' + colorText.END)
             print(tabulate(df, headers='keys', tablefmt='psql'))
-            print(colorText.BOLD + colorText.WARN + "[+] Note: Trend calculation is based on number of recent days to screen as per your configuration." + colorText.END)
-            input(colorText.BOLD + colorText.GREEN + '[+] Press any key to continue..' + colorText.END)
+            print(colorText.BOLD + colorText.WARN +
+                  "[+] Note: Trend calculation is based on number of recent days to screen as per your configuration." + colorText.END)
+            input(colorText.BOLD + colorText.GREEN +
+                  '[+] Press any key to continue..' + colorText.END)
         except FileNotFoundError:
-            print(colorText.BOLD + colorText.FAIL + '[+] Failed to load recently screened result table from disk! Skipping..' + colorText.END)
+            print(colorText.BOLD + colorText.FAIL +
+                  '[+] Failed to load recently screened result table from disk! Skipping..' + colorText.END)
 
-    def isTradingTime():
+    def isTradingTime(self):
         curr = datetime.datetime.now(pytz.timezone('Asia/Kolkata'))
         openTime = curr.replace(hour=9, minute=15)
         closeTime = curr.replace(hour=15, minute=30)
-        isTradingTime = ((openTime <= curr <= closeTime) and
-                        (0 <= curr.weekday() <= 4))
-
+        return ((openTime <= curr <= closeTime) and
+                (0 <= curr.weekday() <= 4))
 
-    def saveStockData(stockDict, configManager):
+    def saveStockData(self, stockDict, configManager):
         today_date = datetime.date.today().strftime("%d%m%y")
         cache_file = "stock_data_" + str(today_date) + ".pkl"
         configManager.deleteStockData(excludeFile=cache_file)
-        if not os.path.exists(cache_file):
+        self.saveDictSize = 0
+        for _ in stockDict:
+            self.saveDictSize += 1
```

Comment:
```suggestion
    self.saveDictSize = len(stockDict)
```
Test with the above modification to avoid looping. This can save some time.

Reply:
Tested, it is working...